doc/devices/dts/device_tree.rst

```diff
@@ -80,6 +80,54 @@ This .fixup file by default resides in the board directory and is named
 dts.fixup.  This fixup file maps the generated include information to the
 current driver/source usage.
 
+.. _dt_vs_kconfig:
+
+Device Tree vs Kconfig
+**********************
+
+As mentioned above there are several tools used to configuring the build with
+Zephyr.
+The two main ones, Device Tree and Kconfig might seem to overlap in purpose,
+but in fact they do not. This section is intended to serve as a reference when
+deciding whether to place a certain configuration item in Device Tree or Kconfig.
+
+The scope of each configuration tool can be summarized as follows:
+
+* Device Tree is used to describe the **hardware** and its **boot-time
+  configuration**.
+* Kconfig is used to describe which **software features** will be built into
+  the final image, and their **configuration**.
```

Comment:
While I agree to this general stating, I think we should clarify the use of 'chosen' node:
```
	chosen {
		zephyr,console = &uart0;
		zephyr,uart-mcumgr = &uart0;
		zephyr,sram = &sram0;
		zephyr,flash = &flash0;
	};
```

Device Tree specification provides the following definition:
"3.5 /chosen Node:
The /chosen node does not represent a real device in the system but describes parameters chosen or specified by the system firmware at run time. It shall be a child of the root node."

This lis kind of SW oriented configuration.

Reply:
That's a good point. Where do you think we should add this information? I would not want to include it in those first 2 lines, but rather add it as a precision later?

Reply:
I added this now a bit below